tools/clang/test/CodeGenHLSL/batch/declarations/precise/matrix.hlsl

```diff
@@ -1,4 +1,4 @@
-// RUN: %dxc -E main -T vs_6_0 %s | FileCheck %s | XFail GitHub #2080
+// RUN: %dxc -E main -T vs_6_0 %s | FileCheck %s
```

Comment:
How could your change fix this???

Reply:
DxilConditionalMem2Reg only promotes allocas if it isn't marked precise.

Reply:
I don't see that in your code? These things should be orthogonal.

Reply:
It's on line 661 of DxilLoopUnroll.cpp in the custom Mem2Reg routine. This was used exclusively for unroll situations but now it's for general uses.